src/middlewared/middlewared/plugins/apps/crud.py

```diff
@@ -0,0 +1,194 @@
+import contextlib
+import errno
+import shutil
+import textwrap
+
+from middlewared.schema import accepts, Dict, returns, Str
+from middlewared.service import CallError, CRUDService, filterable, job
+from middlewared.utils import filter_list
+from middlewared.validators import Match, Range
+
+from .compose_utils import compose_action
+from .ix_apps.lifecycle import add_context_to_values, get_current_app_config, update_app_config
+from .ix_apps.metadata import update_app_metadata
+from .ix_apps.path import get_installed_app_path, get_installed_app_version_path
+from .ix_apps.query import list_apps
+from .ix_apps.setup import setup_install_app_dir
+from .version_utils import get_latest_version_from_app_versions
+
+
+class AppService(CRUDService):
+    class Config:
+        namespace = 'app'
+        datastore_primary_key_type = 'string'
+        cli_namespace = 'app'
+
+    ENTRY = Dict(
+        'app_query',
+        # TODO: Fill this in
```

Comment:
We already have query in place, we should update this as well so it reflects the actual app entry?